NBitcoin.Altcoins/Stratis.cs

```diff
@@ -0,0 +1,517 @@
+﻿using NBitcoin.DataEncoders;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Linq;
+
+namespace NBitcoin.Altcoins
+{
+	public class Stratis : NetworkSetBase
+	{
+		public static Stratis Instance { get; } = new Stratis();
+
+		public override string CryptoCode => "STRAT";
+
+		private Stratis()
+		{
+
+		}
+
+#pragma warning disable CS0618 // Type or member is obsolete
+		public class StratisConsensusFactory : ConsensusFactory
+		{
+			private StratisConsensusFactory()
+			{
+			}
+
+			public static StratisConsensusFactory Instance { get; } = new StratisConsensusFactory();
+
+			public override Block CreateBlock()
+			{
+				return new StratisBlock(new BlockHeader());
+			}
+
+			public override Transaction CreateTransaction()
+			{
+				return new StratisTransaction(this);
+			}
+		}
+
+		public class StratisBlockSignature : IBitcoinSerializable
+		{
+			protected bool Equals(StratisBlockSignature other)
+			{
+				return Equals(signature, other.signature);
+			}
+
+			public override bool Equals(object obj)
+			{
+				if (ReferenceEquals(null, obj)) return false;
+				if (ReferenceEquals(this, obj)) return true;
+				if (obj.GetType() != this.GetType()) return false;
+				return Equals((StratisBlockSignature)obj);
+			}
+
+			public override int GetHashCode()
+			{
+				return (signature?.GetHashCode() ?? 0);
+			}
+
+			public StratisBlockSignature()
+			{
+				this.signature = new byte[0];
+			}
+
+			private byte[] signature;
+
+			public byte[] Signature
+			{
+				get
+				{
+					return signature;
+				}
+				set
+				{
+					signature = value;
+				}
+			}
+
+			internal void SetNull()
+			{
+				signature = new byte[0];
+			}
+
+			public bool IsEmpty()
+			{
+				return !this.signature.Any();
+			}
+
+			public static bool operator ==(StratisBlockSignature a, StratisBlockSignature b)
+			{
+				if (System.Object.ReferenceEquals(a, b))
+					return true;
+
+				if (((object)a == null) || ((object)b == null))
+					return false;
+
+				return a.signature.SequenceEqual(b.signature);
+			}
+
+			public static bool operator !=(StratisBlockSignature a, StratisBlockSignature b)
+			{
+				return !(a == b);
+			}
+
+			#region IBitcoinSerializable Members
+
+			public void ReadWrite(BitcoinStream stream)
+			{
+				stream.ReadWriteAsVarString(ref signature);
+			}
+
+			#endregion
+
+			public override string ToString()
+			{
+				return Encoders.Hex.EncodeData(this.signature);
+			}
+		}
+
+		public class StratisBlock : Block
+		{
+			public StratisBlock(BlockHeader header) : base(header)
+			{
+
+			}
+			public override ConsensusFactory GetConsensusFactory()
+			{
+				return StratisConsensusFactory.Instance;
+			}
+
+			public override void ReadWrite(BitcoinStream stream)
+			{
+				base.ReadWrite(stream);
+				stream.ReadWrite(ref this.blockSignature);
+			}
+
+			public static bool BlockSignature = false;
```

Comment:
Why do you need this boolean? `BlockSignature`

Reply:
I think this was in the original Stratis code. Looks like it has been cleaned up since I wrote this so will take another look but my initial impression is that I should be able to remove it.